tools/disass/radare2/x86.py

```diff
@@ -0,0 +1,167 @@
+import json
+import log
+
+import CFG_pb2
+try:
+    import r2pipe
+except:
+    log.critical("Install the radare2 python bindings. (https://github.com/radare/radare2-r2pipe)")
+
+#
+# I have concerns on performance in large executable scenarios.
+# I do not have faith.. Qapla'
+#
+
+
+#
+# r2 is a r2pipe.open instance
+#
+def analyze(r2):
+
+    # Get list of json dicts that have function information.
+    # An entry in the list is like:
+    # {
+    #   "offset":4196100,
+    #   "name":"sym._fini",
+    #   "size":9,
+    #   "realsz":9,
+    #   "cc":1,
+    #   "nbbs":1,
+    #   "edges":0,
+    #   "ebbs":1,
+    #   "calltype":"amd64",
+    #   "type":"sym",
+    #   "diff":"NEW",
+    #   "callrefs":[{ "address":"", "type" : "", "at" : ""},...],
+    #   "datarefs":[NNN, NNN...],
+    #   "codexrefs":[],
+    #   "dataxrefs":[],
+    #   "difftype":"new",
+    #   "indegree":0,
+    #   "outdegree":0,
+    #   "nargs":0,
+    #   "nlocals":0
+    # }
+
+    #
+    # The purpose of running these is to get the executable
+    # analyzed so the representation given from radare2 is 
+    # in a form that is useful to being remill-ingested.
+    # I am sure there are improvements to be made here.
+    #
+    # Should investigate using:
+    #   aaaa is experimental
+    log.warning("Fuck you")
+    prep_analyses = {
+      "aa" : "alias for 'af@@ sym.*;af@entry0;afva'",
+      "aaa" : " autoname functions after aa (see afna)",
+      "aab" : "analyze blocks over text section",
+      "aac" : "analyze function calls",
+      "aad" : "analyze data references to code",
+      "aae" : "analyze references with ESIL",
+      "aap" : "analyze function preludes"
+    }
+    for x,y in prep_analyses.iteritems():
+        log.info("running command for: {}".format(y))
+        r2.cmd(x)
+    
+    fn_list = r2.cmdj("aflj")
+
+    mod = CFG_pb2.Module()
+    for fne in fn_list:
+        name = fne["name"]
+        new_block = True
+
+        # Having each function be a named block. Is enough?
+        named_block = CFG_pb2.NamedBlock()
+        named_block.name = name
+        named_block.address = fne["offset"]
+
+        # XXX: I am a bit confused on the meaning of the 1 and 0 for
```

Comment:
Use `TODO(name)` format here. That way people know who to ask when they're trying to understand this.